smartsim/_core/control/controller.py

```diff
@@ -346,6 +347,35 @@ def init_launcher(self, launcher: str) -> None:
         else:
             raise TypeError("Must provide a 'launcher' argument")
 
+    @staticmethod
+    def symlink_output_files(
+        job_step: Step, entity: t.Union[SmartSimEntity, EntitySequence[SmartSimEntity]]
+    ) -> None:
+        """Symlink output files under .smartsim to the entity output files
+
+        :param job_step: Job step instance
+        :type job_step: Step
+        :param entity: Entity instance
+        :type entity: SmartSimEntity | EntitySequence[SmartSimEntity]
+        """
+        historical_out, historical_err = job_step.get_output_files()
+        entity_out = osp.join(entity.path, f"{entity.name}.out")
```

Comment:
I _think_ we're moving towards `pathlib` instead of `os.path`. If so and because these are always just local variables, could you change the `entity_out` and `entity_err` to Path objects and update the other `os` operations to use pathlib intrinsics intead.

Reply:
You got it!

Reply:
It looks like entities that call `get_output_files` expect to return strings. Should I start changing all those instances to expect pathlib.Path instead? Or leave it?

Reply:
What I did instead was cast `historical_out` and `historical_err` once they're in the symlink function.